src/plugins/intel_gpu/src/plugin/graph.cpp

```diff
@@ -470,6 +478,15 @@ std::shared_ptr<ngraph::Function> Graph::GetExecGraphInfoByPrimitivesInfo(std::v
 //     [ ov::intel_gpu::Graph::outputDims ]
 //     [ cldnn::network ]
 void Graph::Export(cldnn::BinaryOutputBuffer &ob) {
+#ifdef ENABLE_ONEDNN_FOR_GPU
+    try {
+        get_engine().get_onednn_engine();
+        ob << true;
+    } catch (ov::AssertFailure &) {
+        ob << false;
+    }
+#endif  // ENABLE_ONEDNN_FOR_GPU
+
```

Comment:
Dont we need to handle  such a case that:
1. Model is dumped with openvino built with "ENABLE_ONEDNN_FOR_GPU" => need_onednn_engine flag is stored in the cache 
2. Next, one try to load cache with openvino built without "ENABLE_ONEDNN_FOR_GPU" by mistake => However, since the commit and driver are same, it is trying to use the cache and this data is misread?

Reply:
I updated the logic that checks if OneDNN is required and is enabled in the current gpu plugin, as you commented. Thank you.